### DISCUSSION TITLE
403 forbidden error while updating archived repos

### DISCUSSION THREAD

**Discussion by tgnanakumar** [community/community#24784] [Apps, API and Webhooks]
<p>I’m facing <strong>403 forbidden error</strong> while updating visibility state of archived repos using GitHub API although my personal access token has all the scopes included. However this works fine with my other unarchived repos.</p>
<p>I couldn’t find anything related to this context mentioned in GitHub documentation as well. Any pointers or help in this regard is appreciated.</p>

**Comment by canuckjacq** [comment, marked as answer]
<p>:wave: Welcome!</p>
<p>According to the API documentation for <a href="https://docs.github.com/en/rest/reference/repos#update-a-repository">Update a Repository: archived</a> while you can archive a repository using the API, you can’t unarchive a repository, which would be necessary before <a href="https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-archiving-repositories">any changes can be made to an archived repository</a>, including visibility.</p>
<p>You can of course un-archive a repository using the website in the archived repository’s settings.</p>

**Comment by byrneh** [comment]
<p>Hi <a class="mention" href="/tgnanakumar">@tgnanakumar</a>, I was able to both archive and unarchive a repository using the following GitHub GraphQL APIs, which may allow you to do what your want programmatically, by first archiving the repository, changing visibility as needed and then arching once again,<br>
<a href="https://docs.github.com/en/graphql/reference/mutations#unarchiverepository" rel="noopener nofollow ugc">unarchiverepository</a><br>
<a href="https://docs.github.com/en/graphql/reference/mutations#archiverepository" rel="noopener nofollow ugc">archiverepository</a></p>
<p><a class="mention" href="/canuckjacq">@canuckjacq</a>, I was able to change using the GUI an archived repositories visibility between ‘public’ and ‘private’ whilst it was still archived, this surprised me as its says “This repository has been archived by the owner. It is now read-only.”, it is also inconsistent with the REST API 403 forbidden error mentioned mentioned in this topic, not sure why the GUI and API should behave differently in these circumstances?</p>

**Comment by tgnanakumar** [comment]
<p>Thanks <a class="mention" href="/byrneh">@byrneh</a> for letting me know the possibility of doing it using GraphQL API! I’ll check it out.</p>
